src/operator/controllers/intents_reconcilers/networkpolicy/builders/ingress_network_policy.go

```diff
@@ -0,0 +1,61 @@
+package builders
+
+import (
+	"context"
+	"fmt"
+	"github.com/amit7itz/goset"
+	otterizev1alpha3 "github.com/otterize/intents-operator/src/operator/api/v1alpha3"
+	"github.com/otterize/intents-operator/src/operator/effectivepolicy"
+	"github.com/otterize/intents-operator/src/shared/injectablerecorder"
+	v1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type IngressNetpolBuilder struct {
+	injectablerecorder.InjectableRecorder
+}
+
+func NewIngressNetpolBuilder() *IngressNetpolBuilder {
+	return &IngressNetpolBuilder{}
+}
+
+// a function that builds ingress rules from serviceEffectivePolicy
+func (r *IngressNetpolBuilder) buildIngressRulesFromServiceEffectivePolicy(ep effectivepolicy.ServiceEffectivePolicy) []v1.NetworkPolicyIngressRule {
+	ingressRules := make([]v1.NetworkPolicyIngressRule, 0)
+	fromNamespaces := goset.NewSet[string]()
+	for _, call := range ep.CalledBy {
+		if call.IntendedCall.IsTargetOutOfCluster() {
+			continue
+		}
+		if call.IntendedCall.IsTargetServerKubernetesService() {
+			continue
+		}
+		if fromNamespaces.Contains(call.Service.Namespace) {
+			continue
+		}
+		// We should add only one ingress rule per namespace
```

Comment:
Please add explanation why - it's because there is no difference in the pod selector, just the namespace selector
Or use an implementation that shows this detail, such as dedupping by (podSelector, namespaceSelector) using a map